### PR TITLE
Make barrier fail messages as hud and edit a message

### PR DESCRIPTION
**Pull request by DiamondPlane**

### CODE REVIEW

**Comment by xenonca** [comment]
Need to add `hud_events` to the mods dependencies I think

**Comment by DiamondPlane** [comment]
> Need to add `hud_events` to the mods dependencies I think
Yes

File: mods/ctf/ctf_map/ctf_map_core/barrier.lua [hunk 1]
```diff
@@ -222,7 +222,11 @@ if minetest.get_modpath("ctf") then
 		local tname = ctf.player(name).team
 		if tname and
 				(tname == "blue" and pos.z >= 0) or (tname == "red" and pos.z <= 0) then
-			minetest.chat_send_player(name, "Can't dig beyond the barrier!")
+			hud_event.new(name, {
+			name = "ctf_map:barrier",
```
Comment (xenonca) [inline mods/ctf/ctf_map/ctf_map_core/barrier.lua:226]:
```suggestion
			name = "ctf_map:barrier_dig",
```

File: mods/ctf/ctf_map/ctf_map_core/barrier.lua [hunk 2]
```diff
@@ -222,7 +222,11 @@ if minetest.get_modpath("ctf") then
...
+			color = "0xFFFFFF",
+			value = "Can't dig or place node beyond barrier!"
```
Comment (xenonca) [inline mods/ctf/ctf_map/ctf_map_core/barrier.lua:228]:
```suggestion
			value = "Can't dig or place nodes beyond the barrier!"
```

File: mods/ctf/ctf_map/ctf_map_core/barrier.lua [hunk 3]
```diff
@@ -238,7 +242,11 @@ if minetest.get_modpath("ctf") then
 			if ctf_map.get_team_relative_z(player) < 0 and not ctf_map.can_cross(player) then
 				local name = player:get_player_name()
 				if ctf.move_to_spawn(name) then
-					minetest.chat_send_player(name, "Match hasn't started yet!")
+					hud_event.new(name, {
+					name = "ctf_map:barrier",
```
Comment (xenonca) [inline mods/ctf/ctf_map/ctf_map_core/barrier.lua:246]:
```suggestion
					name = "ctf_map:barrier_buildtime",
```

**Review by xenonca — CHANGES_REQUESTED** [review]

File: mods/ctf/ctf_map/ctf_map_core/barrier.lua (hunk 2 above)
Comment (DiamondPlane) [reply mods/ctf/ctf_map/ctf_map_core/barrier.lua:228]: That's a good idea

**Pull request closed without merging**